sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncToken.java

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.annotation.Immutable;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.logging.ClientLogger;
+
+/**
+ * Due to the nature of some distributed systems real-time consistency between requests can't (or it's very hard) to be
+ * enforced implicitly. A solution is to allow protocol support in the form of multiple Synchronization Tokens.
+ * Synchronization tokens are optional.
+ *
+ * Uses optional Sync-Token request/response headers will guarantee real-time consistency between different client
+ * instances and requests.
+ *
+ * @see <a href="https://github.com/Azure/AppConfiguration/blob/master/docs/REST/consistency.md"/>
+ */
+@Immutable
+public final class SyncToken {
+    private static final ClientLogger LOGGER = new ClientLogger(SyncToken.class);
+
+    private final String id;
+    private final String value;
+    private final long sequenceNumber;
+
+    /**
+     * Create an instance of SyncToken class.
+     *
+     * @param id Token ID (opaque)
+     * @param value Token value (opaque). Allows base64 encoded string
+     * @param sequenceNumber Token sequence number (version). Higher means newer version of the same token.
+     * Allows for better concurrency and client cache ability. The client may choose to use only token's last version,
+     * since token versions are inclusive. Not required for requests.
+     */
+    private SyncToken(String id, String value, long sequenceNumber) {
+        this.id = id;
+        this.value = value;
+        this.sequenceNumber = sequenceNumber;
+    }
+
+    /**
+     * Create an {@code SyncToken} by parsing a given sync-token string with only one sync-token.
+     *
+     * @param syncToken only one raw sync-token string from HTTP response header, ex.,
+     * <p>Sync-Token: {@code <id>=<value>;sn=<sn></p>}. But not
+     * <p>Sync-Token: {@code <id>=<value>;sn=<sn>,<id>=<value>;sn=<sn>}</p>
+     * @return {@code SyncToken} instance
+     */
+    static SyncToken parseSyncToken(String syncToken) {
```

Comment:
Why a static factory method versus a constructor that does the parsing? I don't see any value that this adds.

Reply:
From my point of view, it brings a meaningful method name for parsing syncToken string. So the constructor will not be performing further tasks besides just initializing fields. Functionally, it has no difference to constructor but just easier to test.

I am fine to change to just use constructor if it doesn't look good to you.

Reply:
How is it easier to test? If you want to use a static parse method, it should be throwing an exception if parsing a sync token is not possible rather than returning null.